# Bariantza Analisia: Ongarriaren adibidea

Galsoroen batez besteko etekinerako ongarri motaren eragin posiblea aztertu nahi da. Honetarako, 16 lur-zati homogeneo aukeratu, banaka banatu zoriz eta talde bakoitzekoak ongarri mota berdinaz erein dira. Lorturiko emaitzak  kintale/Ha-kotan izan dira (begiratu 3. gaiko teoriaren 1. adibidea). 



In [2]:
# excel datuak irakurri
 library("readxl")
ongarria <-read_excel("Ongarria.xls")
str(ongarria)
#Ongarrimota aldagaia faktorera aldatuko dugu
ongarria$Ongarrimota.f <-as.factor(ongarria$Ongarrimota)

readxl works best with a newer version of the tibble package.
You currently have tibble v1.4.1.
Falling back to column name repair from tibble <= v1.4.2.
Message displays once per session.



Classes ‘tbl_df’, ‘tbl’ and 'data.frame':	16 obs. of  2 variables:
 $ Etekina    : num  218 198 215 174 263 201 219 226 274 248 ...
 $ Ongarrimota: chr  "A" "A" "A" "A" ...


Hasteko, etekinaren deskribapena egin dezakegu, ongarri motaren arabera: 

In [30]:
#install.packages("RcmdrMisc")
#library("RcmdrMisc")
numSummary(ongarria[,"Etekina"], groups=ongarria$Ongarrimota.f,  statistics=c("mean", "sd", "quantiles"), quantiles=c(0,.25,.5,.75,1))

    mean       sd  0%    25%   50%    75% 100% Etekina:n
A 201.25 20.18869 174 192.00 206.5 215.75  218         4
B 227.25 26.05603 201 214.50 222.5 235.25  263         4
C 272.50 37.54553 243 246.75 261.0 286.75  325         4
D 222.00 65.13064 180 182.25 195.0 234.75  318         4

### Normalitatea egiaztatu

In [9]:
shapiro.test(ongarria$Etekina[ongarria$Ongarrimota=="A"])
ks.test(ongarria$Etekina[ongarria$Ongarrimota=="A"],pnor)


	Shapiro-Wilk normality test

data:  ongarria$Etekina[ongarria$Ongarrimota == "A"]
W = 0.89425, p-value = 0.4031


ERROR: Error in ks.test(ongarria$Etekina[ongarria$Ongarrimota == "A"]): argument "y" is missing, with no default


A ongarri motaren etekinaren normaltasuna aztertzean, Shapiro-wilken testaren estatistikoaren balioa 0.89415 da eta p-balioa 0.4301. Hortaz, $\%5$eko adierazgarritasun mailaz, ez dugu A ongarri motaren etekinaren normaltasuna baztertzen.

#### Oharrak: 
* B, C eta D ongarri moten etekinaren normaltasuna, modu berdinean egiaztatzen da.
* Normaltasuna egiaztatu nahi den laginaren tamaina $30$ baino handiagoa bada, Kolmogorov-Smirnoff erabili behar da (ks.test erabiliz)

### Bariantzen homogeneotasuna

In [32]:
leveneTest(ongarria$Etekina, ongarria$Ongarrimota.f, center=mean) 

Df F value  Pr(>F)   
group  3 1.804026 0.2000296
      12       NA        NA

Leveneren testaren estatistikoaren balioa 1.804026 da, askatasun graduak 3 eta p-balioa 0.2000296. Hortaz, $\%5$eko adierazgarritasun mailaz, ez dugu bariantzen berdintasuna baztertzen.

Beraz  bariantza-analisia  metodo  egokia  da  datu  horiek  aztertzeko.  Kontrastatu  behar  ditugun hipotesiak hauek dira: 

$H_0:\ \mu_A=\mu_B=\mu_C=\mu_D$

$H_1: \exists i,j$ non $\mu_j\neq \mu_i$



In [33]:
Anova <-  aov(Etekina ~ Ongarrimota.f, data=ongarria)
summary(Anova)

              Df Sum Sq Mean Sq F value Pr(>F)
Ongarrimota.f  3  10808    3603   2.139  0.149
Residuals     12  20214    1685               

p-balioa = 0.1486 > 0.05 denez, ez dugu $H_0$ errefusatzen eta, beraz, lau populazioen batez besteko etekinen arteko desberdintasun adierazgarririk ez dago. Orduan, datu hauen arabera, ongarri motak batez besteko etekinean eraginik ez duela esango dugu. 